packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -791,10 +799,31 @@ export async function getAvailableSlots({ input, ctx }: GetScheduleOptions): Pro
         if (isFutureLimitViolationForTheSlot) {
           foundAFutureLimitViolation = true;
         }
+
+        // Before enabling minimum booking notice for seats, the following outOfBounds check was redundant.
+        // Now its needed, because when seats have a minimum booking notice (seatsMinimumBookingNotice),
+        // minimumBookingNotice is not applied in getSlots, so we need to filter out the slots that
+        // are out of bounds here.
+
+        let isOutOfBounds = false;
+        if (eventType.seatsPerTimeSlot && slot.attendees && slot.attendees > 0) {
+          // logic for handling eventType.seatsMinimumBookingNotice if the eventType has seats
+          isOutOfBounds = isTimeOutOfBounds({
+            time: slot.time,
+            minimumBookingNotice: eventType.seatsMinimumBookingNotice ?? undefined,
+          });
+        } else {
+          // default case for outOfBounds as usual, only if no seats for event type
+          isOutOfBounds = isTimeOutOfBounds({
+            time: slot.time,
+            minimumBookingNotice: eventType.minimumBookingNotice,
+          });
+        }
+
         return (
-          !isFutureLimitViolationForTheSlot &&
+          !isFutureLimitViolationForTheSlot && !isOutOfBounds
           // TODO: Perf Optmization: Slots calculation logic already seems to consider the minimum booking notice and past booking time and thus there shouldn't be need to filter out slots here.
-          !isTimeOutOfBounds({ time: slot.time, minimumBookingNotice: eventType.minimumBookingNotice })
+          // && !isTimeOutOfBounds({ time: slot.time, minimumBookingNotice: eventType.minimumBookingNotice })
         );
```

Comment:
Here, we use the existing code that was marked as redundant by the comment. It used to filter out slots that are out of bounds in regard to the minimumBookingNotice. We extended it to handle the case that a seatMinimumBookingNotice is set and a seats booking has existing attendees. With that, we handled that single case where minimumBookingNotice was overwritten with 0 by our seatsMinimumBookingNoticeActive flag when calling getSlots().